shared/test-helpers/assertThrow.js

```diff
@@ -1,26 +1,28 @@
-function assertError(error, s, message) {
-  assert.isAbove(error.message.search(s), -1, `${message}\n\t(error: ${error})`);
+function assertError(error, expectedErrorCode, expectedErrorMessage = '') {
+  assert(error.message.search(expectedErrorCode) > -1, `Expected error code "${expectedErrorCode}" but failed with "${error}" instead.`)
+  assert(error.message.search(expectedErrorMessage) > -1, `Expected error message "${expectedErrorMessage}" but failed with "${error}" instead.`)
 }
 
-async function assertThrows(block, message, errorCode) {
+async function assertThrows(blockOrPromise, expectedErrorCode, expectedErrorMessage = '') {
   try {
-    await block()
-  } catch (e) {
-    return assertError(e, errorCode, message)
+    (typeof blockOrPromise === 'function') ? await blockOrPromise() : await blockOrPromise
+  } catch (error) {
+    assertError(error, expectedErrorCode, expectedErrorMessage)
+    return
   }
-  assert.fail('should have thrown before')
+  assert.fail(`Expected "${expectedErrorCode}" but it did not fail`)
 }
 
 module.exports = {
-  async assertJump(block, message = 'should have failed with invalid JUMP') {
-    return assertThrows(block, message, 'invalid JUMP')
+  async assertJump(blockOrPromise, message = '') {
+    return assertThrows(blockOrPromise, 'invalid JUMP', message)
   },
 
-  async assertInvalidOpcode(block, message = 'should have failed with invalid opcode') {
-    return assertThrows(block, message, 'invalid opcode')
+  async assertInvalidOpcode(blockOrPromise, message = '') {
+    return assertThrows(blockOrPromise, 'invalid opcode', message)
   },
 
-  async assertRevert(block, message = 'should have failed by reverting') {
-    return assertThrows(block, message, 'revert')
+  async assertRevert(blockOrPromise, message = '') {
+    return assertThrows(blockOrPromise, 'revert', message)
```

Comment:
I extended the functionality of this helper to support promises and error messages as well, making sure it is backwards compatible to what we have currently. It basically simplifies the way it can be used (see [here](https://github.com/aragon/aragon-apps/pull/736/commits/f2b94390ac954cf8ce94e0d93f529113fe5971fa#diff-60f24b1826110bb61491ab38fa09ecdcR206)). 

If you guys like it, I can take care and propagate it through the rest of the codebase. If you think this should be part of a separate PR, no problem at all as well :)

Reply:
Would be amazing to do so!

We should also add the ability to specify a specific revert message, but I can do that as part of the changes to Finance.

Reply:
See https://github.com/aragon/aragon-apps/pull/741/commits/ac7eb6a67b18792a06c5babfcddf96862780b505.